packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -60,45 +50,40 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
   protected readonly handlers = {
     // eth_requestAccounts
     handshake: async (_: RequestArguments): Promise<AddressString[]> => {
-      if (this.connected) {
-        this.emit('connect', { chainId: hexStringFromIntNumber(IntNumber(this.chain.id)) });
-        return this.accounts;
+      if (!this.signer) {
```

Comment:
review it with whitespace hidden 🙏 
https://github.com/coinbase/coinbase-wallet-sdk/pull/1331/files?w=1